internal/service/docdb/sweep.go

```diff
@@ -23,6 +23,212 @@ func init() {
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error retrieving DocDB Clusters: %w", err)
+	}
+
+	return nil
+}
+
+func sweepDBClusterSnapshots(region string) error {
+	client, err := sweep.SharedRegionalSweepClient(region)
+
+	if err != nil {
+		return fmt.Errorf("error getting client: %w", err)
+	}
+
+	conn := client.(*conns.AWSClient).DocDBConn
+	input := &docdb.DescribeDBClusterSnapshotsInput{}
+
+	err = conn.DescribeDBClusterSnapshotsPages(input, func(out *docdb.DescribeDBClusterSnapshotsOutput, lastPage bool) bool {
+		for _, dBClusterSnapshot := range out.DBClusterSnapshots {
+			name := aws.StringValue(dBClusterSnapshot.DBClusterSnapshotIdentifier)
+			input := &docdb.DeleteDBClusterSnapshotInput{
+				DBClusterSnapshotIdentifier: dBClusterSnapshot.DBClusterSnapshotIdentifier,
+			}
+
+			log.Printf("[INFO] Deleting DocDB Cluster Snapshot: %s", name)
+
+			_, err := conn.DeleteDBClusterSnapshot(input)
+
+			if err != nil {
+				log.Printf("[ERROR] Failed to delete DocDB Cluster Snapshot (%s): %s", name, err)
+				continue
+			}
+
+			if err := WaitForDBClusterSnapshotDeletion(context.TODO(), conn, name, DBClusterSnapshotDeleteTimeout); err != nil {
+				log.Printf("[ERROR] Failure while waiting for DocDB Cluster Snapshot (%s) to be deleted: %s", name, err)
+			}
+		}
+		return !lastPage
+	})
+
+	if sweep.SkipSweepError(err) {
+		log.Printf("[WARN] Skipping DocDB Cluster Snapshot sweep for %s: %s", region, err)
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error retrieving DocDB Cluster Snapshots: %w", err)
+	}
+
+	return nil
+}
+
+func sweepDBClusterParameterGroups(region string) error {
+	client, err := sweep.SharedRegionalSweepClient(region)
+
+	if err != nil {
+		return fmt.Errorf("error getting client: %w", err)
+	}
+
+	conn := client.(*conns.AWSClient).DocDBConn
+	input := &docdb.DescribeDBClusterParameterGroupsInput{}
+
+	err = conn.DescribeDBClusterParameterGroupsPages(input, func(out *docdb.DescribeDBClusterParameterGroupsOutput, lastPage bool) bool {
+		for _, dBClusterParameterGroup := range out.DBClusterParameterGroups {
+			name := aws.StringValue(dBClusterParameterGroup.DBClusterParameterGroupName)
+			input := &docdb.DeleteDBClusterParameterGroupInput{
+				DBClusterParameterGroupName: dBClusterParameterGroup.DBClusterParameterGroupName,
+			}
+
+			log.Printf("[INFO] Deleting DocDB Cluster Parameter Group: %s", name)
+
+			_, err := conn.DeleteDBClusterParameterGroup(input)
+
+			if err != nil {
+				log.Printf("[ERROR] Failed to delete DocDB Parameter Group (%s): %s", name, err)
+				continue
+			}
+		}
+		return !lastPage
+	})
+
+	if sweep.SkipSweepError(err) {
+		log.Printf("[WARN] Skipping DocDB Cluster Parameter Group sweep for %s: %s", region, err)
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error retrieving DocDB Cluster Parameter Groups: %w", err)
+	}
+
+	return nil
+}
+
+func sweepDBInstances(region string) error {
```

Comment:
Since there will likely be a number of instances, and they can each take some time to destroy, we can make the deletion concurrent. Search for `sweep.SweepOrchestrator` for examples of how to implement this